sklearn/utils/estimator_checks.py

```diff
@@ -1548,16 +1568,16 @@ def check_outliers_train(name, estimator_orig, readonly_memmap=True):
 
     decision = estimator.decision_function(X)
```

Comment:
```python
for func in ['decision_function', 'score_samples']:
    output = getattr(estimator, func)(X)
    assert output.dtype == np.dtype('float')
    assert output.shape == (n_samples,)
```

Reply:
I now use a for loop but a bit different to your suggestion as we need the outputs for other checks.